src/wp-includes/functions.php

```diff
@@ -8458,3 +8458,87 @@ function wp_recursive_ksort( &$array ) {
 	}
 	ksort( $array );
 }
+
+/**
+ * Attempt to detect a VirtualBox environment.
+ *
+ * This attempts all known methods of detecting VirtualBox.
+ *
+ * @since 6.1.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ *
+ * @return bool Whether or not VirtualBox was detected.
+ */
+function is_virtualbox() {
+	global $wp_filesystem;
+	static $is_virtualbox;
+
+	if ( null !== $is_virtualbox ) {
+		return $is_virtualbox;
+	}
+
+	/**
+	 * Filters whether the current environment uses VirtualBox.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param bool Whether the current environment uses VirtualBox. Default false.
+	 */
+	if ( apply_filters( 'is_virtualbox', false ) ) {
+		$is_virtualbox = true;
+		return $is_virtualbox;
+	}
+
+	// Detection via Composer.
+	if ( function_exists( 'getenv' ) && 'virtualbox' === getenv( 'COMPOSER_RUNTIME_ENV' ) ) {
+		$is_virtualbox = true;
+		return $is_virtualbox;
+	}
+
+	$virtualbox_unames = array( 'vvv' );
+
+	// Detection via `php_uname()`.
+	if ( function_exists( 'php_uname' ) && in_array( php_uname( 'n' ), $virtualbox_unames, true ) ) {
+		$is_virtualbox = true;
+		return $is_virtualbox;
+	}
+
+	/*
+	 * Vagrant can use alternative providers.
+	 * This may not be reliable without some additional checks.
+	 */
+	$virtualbox_usernames = array( 'vagrant' );
```

Comment:
Again, I think this is too specific: there's [a lot of variations](https://app.vagrantup.com/boxes/search?order=desc&page=1&provider=virtualbox&q=php&sort=downloads&utf8=%E2%9C%93) that can't be accounted for

Reply:
Agreed - the comment above this condition is there for this exact reason. It's not reliable as-is and it may not be possible to improve its reliability.

Reply:
Testing for VirtualBox has been removed in favor of those environments providing specific fixes for the issue. As far as core/2225 is concerned, adding the pre_move_dir and post_move_dir hooks is sufficient to enable this type of VB-specific fix now and for any potential similar issue in the future.